modules/gdpr_fields/src/Entity/GdprFieldConfigEntity.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\gdpr_fields\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+
+/**
+ * Defines a Flower configuration entity class.
+ *
+ * @ConfigEntityType(
+ *   id = "gdpr_fields_config",
+ *   label = @Translation("GDPR Fields"),
+ *   handlers = {
+ *   },
+ *   config_prefix = "gdpr_fields_config",
+ *   admin_permission = "view gdpr fields",
+ *   entity_keys = {
+ *     "id" = "id"
+ *   },
+ *   links = {
+ *     "add-form" = "/admin/gdpr/fields/add",
+ *     "edit-form" = "/admin/gdpr/fields/{gdpr_fields_config}/edit2",
+ *     "delete-form" = "/admin/gdpr/fields/{gdpr_fields_config}/delete"
+ *   }
+ * )
+ */
+class GdprFieldConfigEntity extends ConfigEntityBase {
+
+  /**
+   * Associative array.
+   *
+   * Each element is keyed by bundle name and contains an array representing
+   * a list of fields.
+   *
+   * Each field is in turn represented as a nested array.
+   *
+   * @var array
+   */
+  public $bundles = [];
+
+  /**
+   * Sets a GDPR field's settings.
+   *
+   * @param string $bundle
+   *   Bundle.
+   * @param string $field_name
+   *   Field.
+   * @param bool $enabled
+   *   Enabled.
+   * @param string $rta
+   *   RTA setting.
+   * @param string $rtf
+   *   RTF setting.
+   * @param string $sanitizer
+   *   Sanitizer to use.
+   * @param string $notes
+   *   Notes.
+   *
+   * @return $this
+   */
+  public function setField($bundle, $field_name, $enabled, $rta, $rtf, $sanitizer, $notes) {
+    $this->bundles[$bundle][$field_name] = [
+      'bundle' => $bundle,
```

Comment:
I think this could be more generic. rather than having to pass every field in as a separate parameter. $bundle and $field_name are still good to force as they are necessary for any update.
```
  public function setField($bundle, $field_name, $values) {

    $values['bundle'] = $bundle;
    $values['name'] = $field_name;

    foreach ($values as $key => $value) {
      $this->bundles[$bundle][$field_name][$key] = $value;
    }
```

This way we can pass anything in with values, either less than all fields or add more fields without having to change the function.

Reply:
Done